### PR TITLE
feat: add a script to fix package json files for loopback-next

### PR DESCRIPTION
**Pull request by raymondfeng**
## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: bin/update-package-json.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  const packages = await project.getPackages();
+  const rootPath = project.rootPath;
+  const rootPkg = fs.readJsonSync(path.join(rootPath, 'package.json'));
+
+  for (const p of packages) {
+    const pkgFile = path.join(p.location, 'package.json');
+    const pkg = fs.readJsonSync(pkgFile);
+    const current = JSON.stringify(pkg, null, 2);
+
+    const isTypescriptPackage = fs.existsSync(
```
Comment (bajtos) [inline bin/update-package-json.js]: TypeScript is typically spelled as two words in CamelCase, see e.g. https://www.typescriptlang.org

```suggestion
    const isTypeScriptPackage = fs.existsSync(
```

File: bin/run-lerna.js [hunk 2]
```diff
@@ -31,9 +32,5 @@ async function run(argv, options) {
...
 
 module.exports = run;
-if (require.main === module) {
-  run(process.argv).catch(err => {
-    console.error(err);
-    process.exit(1);
-  });
-}
+
+main(module, run, process.argv);
```
Comment (bajtos) [inline bin/run-lerna.js:36]: Let's pick a more descriptive name please, one that will make it more clear what is this line going to do.

For example:

```suggestion
runIfMainCliModule(module, run, process.argv);
```

File: bin/script-util.js [hunk 3]
```diff
@@ -83,6 +91,27 @@ function isTypeScriptPackage(pkg) {
   return fs.existsSync(path.join(pkg.location, 'tsconfig.json'));
 }
 
+/**
+ * Run an async function
```
Comment (bajtos) [inline bin/script-util.js:95]: Please improve the comment to describe what is going to happen and under which conditions.

File: bin/script-util.js [hunk 4]
```diff
@@ -83,6 +91,27 @@ function isTypeScriptPackage(pkg) {
...
+ * @param {*} asyncFn - An async function
+ * @param {*} args - Arguments
```
Comment (bajtos) [inline bin/script-util.js:97]: Please add `currentModule` to the list of params.

**Review by bajtos — APPROVED** [review]
Great improvements in our tooling 👏🏻 

It would be great if we could package these tools and publish them to npmjs.org, similarly to how we are publishing `@loopback/build` with tooling dealing with a single package. That way we can use these scripts in our Shopping Example and other LB4 projects.  Example package names for inspiration: `@loopback/monorepo` or `@loopback/monorepo-tooling`? 

Reworking monorepo scripts to a new package is out of scope of this pull request though.

File: bin/run-lerna.js (hunk 2 above)
Comment (raymondfeng) [reply bin/run-lerna.js]: I renamed to `runMain()`

**Comment by raymondfeng** [comment]
> It would be great if we could package these tools and publish them to npmjs.org, similarly to how we are publishing @loopback/build with tooling dealing with a single package. That way we can use these scripts in our Shopping Example and other LB4 projects. Example package names for inspiration: @loopback/monorepo or @loopback/monorepo-tooling?

I was motivated to do so. But the root package.json for a lerna repo cannot support a local package as its dev dependency (a real npm install is performed instead of symbolic links).

**Comment by raymondfeng** [comment]
A follow-up PR: https://github.com/strongloop/loopback-next/pull/6488

**Pull request merged**